xdg-terminal-exec

```diff
@@ -40,267 +32,229 @@ else
-	for CONFIG_DIR in $CONF_HIERARCHY; do
-		for DESKTOP in $DESKTOPS_LC; do
-			printf '%s\n' "${CONFIG_DIR}/${DESKTOP}${DESKTOP:+-}${CONFIG_NAME}"
-		done
-		printf '%s\n' "${CONFIG_DIR}/${CONFIG_NAME}"
-	done | uniq
-	IFS="$OIFS"
-}
-
-find_preferred_entry() {
-	# Read entries listed in config files until a valid one is found
-	CONFIG_LIST="$(generate_config_list)"
-
-	debug 'finding preferred entry in configs'
-
-	while read -r CONFIG_FILE; do
-		debug "looking for config '${CONFIG_FILE}'"
-		if [ -f "${CONFIG_FILE}" ]; then
-			debug "reading config '${CONFIG_FILE}'"
-			while read -r LINE; do
-				debug "parsing line '$LINE'"
-				LINE="$(trim_comment "$LINE")"
-				LINE="$(trim_spaces "$LINE")"
-				[ -z "$LINE" ] && continue || ENTRY_ID="$LINE"
-				debug "retrieved entry ID '$ENTRY_ID'"
-				check_entry_id "$ENTRY_ID" || continue
-
-				debug "finding path for entry ID '$ENTRY_ID'"
-				ENTRY_PATH="$(find_entry_path "$ENTRY_ID")"
-				if [ -n "$ENTRY_PATH" ] && check_entry_path "$ENTRY_PATH"; then
-					debug "file at path '$ENTRY_PATH' checks out"
-					printf '%s' "$ENTRY_PATH"
-					return 0
-				else
-					debug "entry ID '$ENTRY_ID' failed check at path '$ENTRY_PATH', blacklisting"
-					BLACKLIST="${BLACKLIST}${BLACKLIST:+;}$ENTRY_ID"
-					continue
-				fi
-			done < "${CONFIG_FILE}"
-		fi
-	done <<- EOF
-		$CONFIG_LIST
-	EOF
+trim_start() {
+	# Remove all but leading whitespace, and trim that from the given string
+	printf '%s' "${1#"${1%%[![:space:]]*}"}"
 }
 
-find_any_entry() {
-	# Read entries in data dirs until a valid one is found
-	debug 'looking for first available entry in data hierarchy'
-	IFS=':'
-	for DATA_DIR in $DATA_HIERARCHY; do
-		IFS="$OIFS"
-		if [ -d "${DATA_DIR}/${DATA_PREFIX_DIR}" ]; then
-			debug "searching in '${DATA_DIR}/${DATA_PREFIX_DIR}'"
-			ENTRY_FILES="$(find -L "${DATA_DIR}/${DATA_PREFIX_DIR}" -type f -iname '*.desktop')"
-			while read -r ENTRY_PATH; do
-				[ -z "$ENTRY_PATH" ] && continue
-				ENTRY_FILE="${ENTRY_PATH#"${DATA_DIR}/${DATA_PREFIX_DIR}/"}"
-				ENTRY_ID="$(printf '%s' "$ENTRY_FILE" | tr '/' '-')"
-				check_entry_id "$ENTRY_ID" || continue
-				if check_entry_path "$ENTRY_PATH"; then
-					debug "file at path '$ENTRY_PATH' checks out"
-					printf '%s' "$ENTRY_PATH"
-					return 0
-				else
-					debug "entry ID '$ENTRY_ID' failed check at path '$ENTRY_PATH', blacklisting"
-					BLACKLIST="${BLACKLIST}${BLACKLIST:+;}$ENTRY_ID"
-					debug "current blacklist: '$BLACKLIST'"
-				fi
-			done <<- EOF
-				$ENTRY_FILES
-			EOF
-		fi
-	done
-	IFS="$OIFS"
+# Read config from given path, print sanitised entries
+read_config_path() {
+	config_path="$1"
+
+	debug "checking path '$config_path'"
+	[ -f "$config_path" ] || return 0
+	# Let `read` trim leading/trailing whitespace from the line
+	while read -r line; do
+		debug "parsing config line '$line'"
+		case $line in
+		# Comment, skip
+		'#'*) ;;
+		# Add a trailing space, so command substitution doesn't trim trailing newlines
```

Comment:
Do you have anything against appending newline-separated lists with `LIST=${LIST}${LIST:+${NEWLINE}}${ITEM}`? Seems more explicit without having to worry about leaving and accounting for newlines/spaces in distant functions

Reply:
I don't, I think I just got carried away. I'll change it

Reply:
Done